src/main/java/BotChat.java

```diff
@@ -0,0 +1,160 @@
+    private static Pattern unmarkPattern = Pattern.compile("unmark \\d+");
+
+    public static void addTask(String s) throws Exception {
+        String taskDescription;
+        switch (parser.extractCommand(s)) {
+            case "todo":
+                taskDescription = parser.extractDescription(s);
+                Task newDescriptionTask = new ToDo(taskDescription);
+                storage.addToDataStore(newDescriptionTask);
+                taskArrayList.addTask(newDescriptionTask);
+                break;
+            case "deadline":
+                taskDescription = parser.extractDescription(s);
+                String[] deadlineStringParts = taskDescription.split("/by ");
+                if (deadlineStringParts.length < 2) {
+                    throw new IncompleteCommandException("task.Deadline command incomplete. It should be in the form of " +
+                            "deadline description /by datetime.");
+                } else {
+                    Task newDeadlineTask = new Deadline(deadlineStringParts[0], deadlineStringParts[1]);
+                    storage.addToDataStore(newDeadlineTask);
+                    taskArrayList.addTask(newDeadlineTask);
+                }
+                break;
+            case "event":
+                taskDescription = parser.extractDescription(s);
+                String[] eventStringParts = taskDescription.split("/from |/to ");
+                if (eventStringParts.length < 3) {
+                    throw new IncompleteCommandException("task.Event command incomplete. It should be in the form of " +
+                            "event description /from datetime /to datetime.");
+                } else {
+                    Task newEventTask = new Event(eventStringParts[0], eventStringParts[1], eventStringParts[2]);
+                    storage.addToDataStore(newEventTask);
+                    taskArrayList.addTask(newEventTask);
+                }
+                break;
+            default:
+                throw new InvalidCommandException(s);
+        }
+    }
+
+    public static String response(String s) {
+        Matcher markMatcher = markPattern.matcher(s);
+        Matcher unmarkMatcher = unmarkPattern.matcher(s);
+        String command = parser.extractCommand(s);
+        if (command.equals("bye")) {
+            terminate = true;
+            return Ui.byeMessage();
+        } else if (command.equals("list")) {
+            StringBuilder stringBuilder = new StringBuilder("Here are the tasks in your list: \n");
+            for (int i = 1; i <= taskArrayList.getLastIdx(); i++) {
+                stringBuilder.append(i);
+                stringBuilder.append(". ");
+                stringBuilder.append(taskArrayList.getTaskByIdx(i - 1).toString());
+                stringBuilder.append("\n ");
+            }
+            return stringBuilder.toString();
+        } else if (markMatcher.matches()) {
+            try {
```

Comment:
To improve readability, I think you can create methods for each case instead of writing the code here.